plugins/output/adapt/outputHelpers.js

```diff
@@ -277,6 +277,63 @@ function getPluginFrameworkVersionCategory(serverFrameworkVersion, pluginMetaDat
   });
 };
 
+function validateCourse(data, cb) {
+  let errors = '';
+  let contentObjects = data.contentobject;
+  let articles = data.article;
+  let blocks = data.block;
+  let components = data.component;
+
+  if (typeof contentObjects === 'undefined') {
+    let courseString = app.polyglot.t('app.course');
+    errors += app.polyglot.t('app.doesnotcontain', {
+      type: courseString[0].toUpperCase() + courseString.slice(1),
+      title: data.course[0].title,
+      childType: app.polyglot.t('app.page', 0)
+    }) + '\n';
+    return cb(errors, false);
+  }
+
+  errors += iterateThroughChildren(contentObjects, articles);
+  errors += iterateThroughChildren(articles, blocks);
+  errors += iterateThroughChildren(blocks, components);
+
+  if (errors.length !== 0) return cb(errors, false);
+
+  return cb(null, true);
+}
+
+function iterateThroughChildren(parents, children) {
+  let errors = '';
+  if (typeof parents === 'undefined') return errors;
+
+  parents.forEach(parent => {
+    let parentType = app.polyglot.t('app.' + parent._type, 1);
+    let childType = app.polyglot.t('app.children');
+
+    if (typeof children === 'undefined') {
+      errors += app.polyglot.t('app.doesnotcontain', {
+        type: parentType[0].toUpperCase() + parentType.slice(1),
+        title: parent.title,
+        childType: childType
+      }) + '\n';
```

Comment:
Since the new code is identical to lines 327–331, suggest storing in a variable that can be called:
```js
const appendError = () => {
  errors += app.polyglot.t('app.doesnotcontain', {
    type: parentType[0].toUpperCase() + parentType.slice(1),
    title: parent.title,
    childType: childType
  }) + '\n';
};
```